modules/migration/uploader.go

```diff
@@ -17,6 +17,14 @@ type Uploader interface {
 	CreateComments(comments ...*Comment) error
 	CreatePullRequests(prs ...*PullRequest) error
 	CreateReviews(reviews ...*Review) error
+	UpdateTopics(topic ...string) error              // update topics of a repository, and delete unused ones
+	UpdateMilestones(milestones ...*Milestone) error // update milestones of a repository, and delete unused ones
+	UpdateLabels(labels ...*Label) error             // rewrite all issue labels and delete unused ones
```

Comment:
> Do they delete every label/milestone/… that is not present in the arguments?

Yes, that's the behavior right now.

Because there's no way to know what's deleted upstream. For Issues and PRs, they are add only, so it makes sense to keep everything. But topics, milestones, labels are different. So to delete things in Gitea, I can only delete anything not present in the arguments. And I think that's the behavior most people want. e.g. if you delete a milestone on GitHub, you would expect it to be deleted on Gitea as well, and don't want to go to Gitea again to delete the same thing.

In the future, we probably can support two-way syncing, but that's not the scope of this PR.